app/bundles/CampaignBundle/EventListener/CampaignSubscriber.php

```diff
@@ -50,13 +60,71 @@ public function __construct(IpLookupHelper $ipLookupHelper, AuditLogModel $audit
     public static function getSubscribedEvents()
     {
         return [
-            CampaignEvents::CAMPAIGN_POST_SAVE   => ['onCampaignPostSave', 0],
-            CampaignEvents::CAMPAIGN_POST_DELETE => ['onCampaignDelete', 0],
-            CampaignEvents::CAMPAIGN_ON_BUILD    => ['onCampaignBuild', 0],
+            CampaignEvents::CAMPAIGN_POST_SAVE         => ['onCampaignPostSave', 0],
+            CampaignEvents::CAMPAIGN_POST_DELETE       => ['onCampaignDelete', 0],
+            CampaignEvents::CAMPAIGN_ON_BUILD          => ['onCampaignBuild', 0],
+            CampaignEvents::ON_CAMPAIGN_TRIGGER_ACTION => ['onCampaignTriggerAction', 1],
         ];
     }
 
     /**
+     * @param CampaignExecutionEvent $event
+     *
+     * @return CampaignExecutionEvent
+     */
+    public function onCampaignTriggerAction(CampaignExecutionEvent $event)
+    {
+        if (!$event->checkContext('campaign.remoteurl')) {
+            return;
+        }
+        $config  = $event->getConfig();
+        $timeout = 10;
+        $headers = [];
+        if (!empty($config['authorization_header'])) {
+            if (strpos($config['authorization_header'], ':') !== false) {
+                list($key, $value) = explode(':', $config['authorization_header']);
+            } else {
+                $key   = 'Authorization';
+                $value = $config['authorization_header'];
+            }
+            $headers[trim($key)] = trim($value);
+        }
+
+        try {
+            $method = $config['method'];
+            $data   = !empty($config['additional_data']['list']) ? $config['additional_data']['list'] : '';
+            $data   = array_flip(AbstractFormFieldHelper::parseList($data));
+            if (in_array($method, ['get', 'trace'])) {
+                $response = $this->connector->$method(
+                    $config['url'],
+                    $headers,
+                    $timeout
+                );
+            } elseif (in_array($method, ['post', 'put', 'patch'])) {
+                $response = $this->connector->$method(
+                    $config['url'],
+                    $data,
+                    $headers,
+                    $timeout
+                );
+            } elseif ($method == 'delete') {
+                $response = $this->connector->$method(
+                    $config['url'],
+                    $headers,
+                    $timeout,
+                    $data
+                );
+            }
+            if (in_array($response->code, [200, 201])) {
+                return $event->setResult(true);
+            }
+        } catch (\Exception $e) {
+        }
```

Comment:
We have to let the users know why the request failed. It fails silently without any clue of what happened. Whether it returned 500 or timeouted. Could you please add some error messages so the users won't report errors like "the campaign URL call doesn't work", but give them information so they could debug themselves?

You can inspire here: https://github.com/mautic/mautic/blob/staging/app/bundles/PluginBundle/EventListener/CampaignSubscriber.php#L116